cli/azd/pkg/executil/builder.go

```diff
@@ -0,0 +1,76 @@
+package executil
+
+import "context"
+
+type Builder struct {
+	commandRunnerFn RunCommandFn
+	runArgs         RunArgs
+}
+
+// Creates a new builder with commandRunnerFn, cmd and args
+func NewBuilder(cmd string, args ...string) *Builder {
+	runArgs := NewRunArgs(cmd, args...)
+
+	return &Builder{
+		runArgs: runArgs,
+	}
+}
+
+// Updates the commandRunnerFn that will used to execute commands
+func (b *Builder) WithCommandRunner(commandRunnerFn RunCommandFn) *Builder {
+	b.commandRunnerFn = commandRunnerFn
+	return b
+}
```

Comment:
I think this is unused given the parameter in `Exec()`

Reply:
Makes sense - will deprecate.